lib/src/rules/use_getters_to_access_properties.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.use_getters_to_access_properties;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+import 'package:linter/src/util/side_effect_visitor.dart';
+
+const _desc =
+    r'Use getters for operations that conceptually access properties.';
+
+const _details = r'''
+
+**DO** use getters for operations that conceptually access properties.
+
+**BAD:**
+```
+rectangle.getWidth()
+collection.isEmpty()
+button.canShow()
+```
+
+**GOOD:**
+```
+rectangle.width
+collection.isEmpty
+button.canShow
+```
+
+''';
+
+bool _beginsWithAsOrTo(String name) {
+  final regExp = new RegExp(r"(to|as|_to|_as)[A-Z]", caseSensitive: true);
+  return regExp.matchAsPrefix(name) != null;
+}
+
+bool _hasInheritedMethod(MethodDeclaration node) =>
+    DartTypeUtilities.lookUpInheritedMethod(node) != null;
+
+bool _hasSideEffect(FunctionBody node) {
+  final visitor = new SideEffectVisitor();
+  node.accept(visitor);
+  return visitor.hasSideEffect;
+}
+
+class UseGettersToAccessProperties extends LintRule {
+  _Visitor _visitor;
+  UseGettersToAccessProperties()
+      : super(
+            name: 'use_getters_to_access_properties',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  visitMethodDeclaration(MethodDeclaration node) {
+    final body = node.body;
+    if (!node.isSetter &&
+        !node.isGetter &&
+        !_hasInheritedMethod(node) &&
+        node.parameters?.parameters?.isEmpty == true &&
+        node.returnType?.type?.name != 'void' &&
+        !(body is BlockFunctionBody && body.block.statements.isEmpty) &&
+        !_beginsWithAsOrTo(node.name.name) &&
+        !_hasSideEffect(node.body)) {
```

Comment:
I think this should also exclude methods with an EmptyFunctionBody, as per earlier comments.

Also, `node.body` can be just `body`.